src/lambda/commands/deleteLambda.ts

```diff
@@ -5,37 +5,63 @@
 
 'use strict'
 
-import * as nls from 'vscode-nls'
-const localize = nls.loadMessageBundle()
+import { OutputChannel, window } from 'vscode'
+import { loadMessageBundle } from 'vscode-nls'
+const localize = loadMessageBundle()
 
 import { LambdaClient } from '../../shared/clients/lambdaClient'
-import { ext } from '../../shared/extensionGlobals'
 import { StandaloneFunctionNode } from '../explorer/standaloneNodes'
 
-export async function deleteLambda(
-    node: StandaloneFunctionNode,
-    refresh: () => void
-): Promise<void> {
-    const client: LambdaClient = ext.toolkitClientBuilder.createLambdaClient(node.regionCode)
+export async function deleteLambda({
+                           lambdaClient,
+                           node,
+                           outputChannel,
+                           onRefresh,
+                           onConfirm = async () => {
+                               const responseNo: string = localize('AWS.generic.response.no', 'No')
+                               const responseYes: string = localize('AWS.generic.response.yes', 'Yes')
+                               const response = await window.showWarningMessage(
+                                 localize(
+                                   'AWS.command.deleteLambda.confirm',
+                                   "Are you sure you want to delete lambda function '{0}'?",
+                                   node.configuration.FunctionName
+                                 ),
+                                 responseYes,
+                                 responseNo
+                               )
+
+                               return response === responseYes
+                           },
+                           onError = (error: any) => {
```

Comment:
Similarly to `onConfirm`, we want to mock the output channel without losing test coverage of the rest of the error handler.